src/test/java/com/puppycrawl/tools/checkstyle/bdd/InputConfiguration.java

```diff
@@ -75,7 +75,10 @@ public String getCheckName() {
     }
 
     public DefaultConfiguration createConfiguration() {
-        return new DefaultConfiguration(checkName);
+        final DefaultConfiguration parsedConfig = new DefaultConfiguration(checkName);
+        defaultProperties.forEach(parsedConfig::addAttribute);
```

Comment:
only `nonDefault` should be added

Reply:
Same reason https://github.com/checkstyle/checkstyle/pull/10106#discussion_r654936865

Reply:
done